qiime2/core/testing/pipeline.py

```diff
@@ -195,6 +195,48 @@ def internal_fail_pipeline(ctx, ints1, ints2, int1, string, fail=False):
     return ints1_ret, ints2_ret, int1_ret
 
 
+def de_facto_list_pipeline(ctx, kwarg=False, non_proxies=False):
+    returns_int = ctx.get_action('dummy_plugin', 'returns_int')
+    list_of_ints = ctx.get_action('dummy_plugin', 'list_of_ints')
+    num_ints = 3
+
+    ints = []
+    for i in range(num_ints):
+        ints_ret, = returns_int(i)
+        ints.append(ints_ret)
+
+    if non_proxies:
```

Comment:
Just for my own understanding, why do you add 1 to `num_ints`?

Reply:
The +1 here is from a previous implementation, and I'm actually going to get rid of it. The way I was doing things before, the ints list would have been [1,2,3] so I needed the +1 to append a 4 instead of another 3. Not like it would break anything to append another 3, I just wanted a 4. Now though this will make [0,1,2,4] which is lame.